internal/batches/service/remote.go

```diff
@@ -98,13 +113,122 @@ func (svc *Service) CreateBatchSpecFromRaw(
 		"allowIgnored":     allowIgnored,
 		"allowUnsupported": allowUnsupported,
 		"noCache":          noCache,
+		"batchChange":      batchChange,
 	}).Do(ctx, &resp); err != nil || !ok {
 		return "", err
 	}
 
 	return resp.CreateBatchSpecFromRaw.ID, nil
 }
 
+// UploadMounts uploads file mounts to the server.
+func (svc *Service) UploadMounts(workingDir string, batchSpecID string, steps []batches.Step) error {
+	if err := svc.areServerSideBatchChangesSupported(); err != nil {
+		return err
+	}
+
+	body := &bytes.Buffer{}
```

Comment:
Also, this is a single request for all uploads, in one multipart body, correct? 

If so, this will be problematic for larger workspace uploads. Batch changes and code intel ran into that and for batch changes that meant uploading each changeset spec individually and then stitching the batch specs together once finished (because diffs can become quite huge we had to learn), and for code intel they implemented chunked uploading which in the backend stitches together larger uploads since lsif uploads can be large singular blobs (like a go binary, so not too far off of our use-case). They also have some mechanism to expire partial uploads at some stage once they are too old, to make sure we don't hoard partial data forever. Link to uploader code: https://sourcegraph.com/github.com/sourcegraph/sourcegraph-incremental-indexing-fork/-/blob/lib/codeintel/upload/upload.go?L18

Reply:
I wrote this function to be generic (so I could upload x files). I want to limit the size of the files that can be upload and the number of files per request (se we are not trying to upload 100s of files in a single request).

One of the ideas I have been wrestling with when I did it this way is bulk upload files vs concurrent upload (1 file per request and send in parallel).

Work still needs to be done on performance... So this probably will change